R/setup_source.R

```diff
@@ -4,64 +4,59 @@
 #' whenever you push commits to GitHub.
 #' It also adds a
 #' [CC-BY 4.0](https://creativecommons.org/licenses/by/4.0/) license file,
-#' a [CODE_OF_CONDUCT.md](https://inbo.github.io/checklist/CODE_OF_CONDUCT.html)
+#' a
+#' [`CODE_OF_CONDUCT.md`](https://inbo.github.io/checklist/CODE_OF_CONDUCT.html)
 #' and the checklist configuration file (`checklist.yml`).
 #' @param path The path to the project.
 #' Defaults to `"."`.
+#' @inheritParams create_package
 #' @export
 #' @importFrom assertthat assert_that
 #' @importFrom desc desc
+#' @importFrom fs dir_create file_copy path path_real
 #' @importFrom gert git_add
 #' @importFrom utils file_test
 #' @family setup
-setup_source <- function(path = ".") {
-  path <- normalizePath(path, winslash = "/", mustWork = TRUE)
+setup_source <- function(path = ".", language) {
```

Comment:
Check in this function if argument `language` is set, or capture and drop all errors from `checklist$new()` here to make errors look like they come from function `setup_source()` itself.  It is rather confusing to see an error coming from `checklist$new()` which is unknown to the user (and even more confusing if the debugging mode is suddenly bringing you somewhere if you are not used to work with it, which happened in my case while I certainly did not do anything myself to activate it).

Reply:
And additionally, as argument `language` is obligatory, it should better be placed before argument `path` (which has a default)

Reply:
Setting `language = "en-GB"` is probably the easiest solution. I'm planning to replace the rigid `setup_source()` with the more flexible `setup_project()`.

Reply:
Eh, in packages this would probably be the correct default in 90% of the cases, but for source projects (mainly INBO reports?) I am not sure.  But as `setup_project()` asks for the language, depreciating `setup_source()` is probably the best solution.;-)

Reply:
Indeed, if you deprecate it in this version as well, it doesn't matter what you choose.  (If I remember well, argument language wasn't there yet in the previous version of checklist.  So maybe it is also a bit strange: the function is updated, and immediately afterwards deprecated. ;-) )

Reply:
fixed in 13fa20e73539